src/Mod/PartDesign/App/FeatureBoolean.cpp

```diff
@@ -178,13 +213,18 @@ void Boolean::handleChangedPropertyName(Base::XMLReader &reader, const char * Ty
     }
 }
 
-TopoDS_Shape Boolean::refineShapeIfActive(const TopoDS_Shape& oldShape) const
+
+// FIXME:  This method ( and the Refine property it depends on ) is redundant with the exact same
+//  thing in FeatureAddSub, but cannot reasonably be moved up an inheritance level to Feature as
+//  there are inheritors like FeatureBox for which a refine Property does not make sense.  A
+//  solution like moving Refine and refineShapeIfActive to a new FeatureRefine class that sits
+//  between Feature and FeatureBoolean / FeatureAddSub is a possibility, or maybe [ew!] hiding the
+//  property in Feature and only enabling it in the places it is relevant.
+TopoShape Boolean::refineShapeIfActive(const TopoShape& oldShape) const

```

Comment:
@wwmayer do you have a preference here? It seems to me that adding a FeatureRefine class is the most reasonable of the options, but I'm not wedded to that approach.

Reply:
> It seems to me that adding a FeatureRefine class is the most reasonable of the options

Seems like most proper solution. There is also the class PartDesign::Transformed that has the method Transformed::refineShapeIfActive which is identical to FeatureAddSub::refineShapeIfActive.

However, the implementation of Boolean::refineShapeIfActive slightly differs from the two implementation above but I think it should be identical to them.

Reply:
This will be resolved in #15690 which might make sense post 1.0.  However, it further uncovered that FeatureTransformed,  FeatureDraft, and FeatureThickness all need further TNP review, which will likely by 1.0 critical.  I don't think any of that holds up this PR.